doc/user-manual/language/cubical.lagda.rst

```diff
@@ -1,106 +1,930 @@
 
-Elements of the interval form a `De Morgan algebra <https://en.wikipedia.org/wiki/De_Morgan_algebra>`_,
-with minimum (``∧``), maximum (``∨``) and negation (``~``).
+
+Partial elements
+----------------
+
+In order to describe the homogeneous composition operations we need to
+be able to write partially specified n-dimensional cubes (i.e. cubes
+where some faces are missing). Given an element of the interval ``r :
+I`` there is a predicate ``IsOne`` which represents the constraint ``r
+= i1``. This comes with a proof that ``i1`` is in fact equal to ``i1``
+called ``1=1 : IsOne i1``. We use Greek letters like ``φ`` or ``ψ``
+when such an ``r`` should be thought of as being in the image of
+``IsOne``.
+
+Using this we introduce a type of partial elements called ``Partial φ
+A``, this is a special version of ``IsOne φ → A`` with a more
+extensional judgmental equality (two elements of ``Partial φ A`` are
+considered equal if they represent the same subcube, so the faces of
+the cubes can for example be given in different order and the two
+elements will still be considered the same). The idea is that
+``Partial φ A`` is the type of cubes in ``A`` that are only defined
+when ``IsOne φ``.  There is also a dependent version of this called
+``PartialP φ A`` which allows ``A`` to be defined only when ``IsOne
+φ``.
 
 .. code-block:: agda
 
-  module interval-example₁ (i j : I) where
-    data _≡_ (i : I) : I → Set where
-      reflI : i ≡ i
+  Partial : ∀ {ℓ} → I → Set ℓ → Setω
+
+  PartialP : ∀ {ℓ} → (φ : I) → Partial φ (Set ℓ) → Setω
+
+Partial elements are introduced using pattern matching:
+
+::
+
+  partialBool : ∀ i → Partial (i ∨ ~ i) Bool
+  partialBool i (i = i0) = true
+  partialBool i (i = i1) = false
 
-    infix 10 _≡_
+The term ``partialBool`` should be thought of a boolean with different
+values when ``(i = i0)`` and ``(i = i1)``. Terms of type ``Partial φ
+A`` can also be introduced using pattern matching lambdas
+(http://wiki.portal.chalmers.se/agda/pmwiki.php?n=ReferenceManual.PatternMatchingLambdas).
 
-    max min neg : I
+::
+
+  partialBool' : ∀ i → Partial (i ∨ ~ i) Bool
+  partialBool' i = λ { (i = i0) → true
+                     ; (i = i1) → false }
+
+When the cases overlap they must agree (note that the order of the
+cases doesn't have to match the interval formula exactly):
+
+::
+
+  partialBool'' : ∀ i j → Partial (~ i ∨ i ∨ (i ∧ j)) Bool
+  partialBool'' i j = λ { (i = i1)          → true
+                        ; (i = i1) (j = i1) → true
+                        ; (i = i0)          → false }
+
+Furthermore ``IsOne i0`` is actually absurd
+
+::
+
+  empty : {A : Set} → Partial i0 A
+  empty = λ { () }
+
+Cubical Agda also has cubical subtypes as in the CCHM type theory:
+
+::
+
+  _[_↦_] : ∀ {ℓ} (A : Set ℓ) (φ : I) (u : Partial φ A) → Setω
+  A [ φ ↦ u ] = Sub A φ u
+
+A term ``v : A [ φ ↦ u ]`` is of type ``A`` and when ``IsOne φ`` it
+must be definitionally equal to ``u : A``. Any term ``u : A`` can be
```

Comment:
It’s not true that `v` is of type `A` (you need to use `ouc` for that), so I would rather say:

A term `v : A [ φ ↦ u ]` should be thought of as a term of type `A` which is definitionally equal to `u:A` when `IsOne φ` is satisfied.

Reply:
Fixed